debian/changelog

```diff
@@ -1,10 +1,13 @@
-cloud-init (22.4-0ubuntu0~22.10.2) UNRELEASED; urgency=medium
+cloud-init (23.1-0ubuntu0~22.10.1) kinetic; urgency=medium
```

Comment:
What we need to add to release documents (and be aware of). Is whether or not our previous release 22.4 had a hot-fix release branch `upstream/ubuntu/kinetic-22.4.x`. If it did, we need to manually sync the top-most debian/changelog entries which don't exist yet in `upstream/ubuntu/devel` which were already officially published. If we don't sync those released changelog entries our diff looks confusing to SRU reviewers (and essentially redacts content that was already publicly visible for known releases).

So, when SRU-ing a new release, prior to new_upstream_snapshot.py, we:
- [ ] check for the presence of a hot fix branch for the previous release `upstream/ubuntu/<SERIES>-22,4,x` 
- [ ] validate any d/changelog and manually sync it `git diff upstream/ubuntu/<SERIES>-22.4.x debian/changelog`
- [ ] validate any Changelog and manually sync it `git diff upstream/ubuntu/<SERIES>-22.4.x Changelog`

Manually add the missing entry either via git cherry-pick or cut-n-paste (watch for trailing whitespace).  git commit -m 'sync d/changelog from hot fix branch ubuntu/kinetic-22.4.x` debian/changelog. Then git rebase -i HEAD~3 and place that commit be

There is a diff telling us we are missing `cloud-init (22.4.2-0ubuntu0~22.10.1) kinetic; urgency=medium` top-most changelog section
```diff
csmith@downtown:/tmp/cloud-init ((d536a5a1...))$ git diff upstream/ubuntu/kinetic-22.4.x debian/changelog 
diff --git a/debian/changelog b/debian/changelog
index c5368242..a3a5bf8f 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,11 +1,13 @@
-cloud-init (22.4.2-0ubuntu0~22.10.1) kinetic; urgency=medium
+cloud-init (23.1-0ubuntu0~22.10.1) kinetic; urgency=medium
 
-  * Upstream snapshot based on 22.4.2 upstream release. (LP: #1996645)
-    - List of changes from upstream can be found at
-      https://raw.githubusercontent.com/canonical/cloud-init/22.4.2/ChangeLog
-    - Includes (LP: #1997559, #1844191) not present in 22.4.0.
+  * d/patches/retain-netplan-world-readable.patch:
+    - Retain original world-readable perms of /etc/netplan/50-cloud-init.yaml.
+      Lunar made the config root read-only.
+  * Upstream snapshot based on 23.1. (LP: #2008230).
+    List of changes from upstream can be found at
+    https://raw.githubusercontent.com/canonical/cloud-init/23.1/ChangeLog
 
- -- James Falcon <james.falcon@canonical.com>  Mon, 28 Nov 2022 10:45:45 -0600
+ -- Alberto Contreras <alberto.contreras@canonical.com>  Thu, 23 Feb 2023 12:32:59 +0100
 
 cloud-init (22.4-0ubuntu0~22.10.1) kinetic; urgency=medium
 ```

Reply:
I pushed https://github.com/blackboxsw/cloud-init/tree/ubuntu/kinetic  as comparative reference

Reply:
in the case of the ChangeLog in the hotfix branches, we've already sync'd those changes into tip of main/ChangeLog so no need to pull in those diffs into the packaging branch as they get pulled in by new_upstream_snapshot.py anyway.